src/blueapi/service/main.py

```diff
@@ -140,6 +142,50 @@ def get_state(handler: Handler = Depends(get_handler)) -> WorkerState:
     return handler.worker.state
 
 
+# Map of current_state: allowed new_states
+_ALLOWED_TRANSITIONS: Dict[WorkerState, Set[WorkerState]] = {
+    WorkerState.RUNNING: {WorkerState.PAUSED},
+    WorkerState.PAUSED: {WorkerState.RUNNING},
+}
+
+
+@app.put(
+    "/worker/state",
+    status_code=status.HTTP_400_BAD_REQUEST,
+    responses={
+        status.HTTP_400_BAD_REQUEST: {"detail": "Transition not allowed"},
+        status.HTTP_202_ACCEPTED: {"detail": "Transition requested"},
+    },
+)
+def set_state(
+    state_change_request: StateChangeRequest,
+    response: Response,
+    handler: Handler = Depends(get_handler),
+) -> WorkerState:
+    """
+    Request that the worker is put into a particular state.
+    Returns the state of the worker at the end of the call.
+    If the worker is PAUSED:
+        new_state may be RUNNING to resume.
+    If the worker is RUNNING:
+        new_state may be PAUSED to pause.
+        defer may be True to defer the pause until the new checkpoint.
+    All other values of new_state will result in 400 "Bad Request"
+    """
+    current_state = handler.worker.state
+    new_state = state_change_request.new_state
+    if (
+        current_state in _ALLOWED_TRANSITIONS
+        and new_state in _ALLOWED_TRANSITIONS[current_state]
+    ):
```

Comment:
If this is false should there be some handling to return the 400?

Reply:
Ah does the response default to 400 due to 
```
status_code=status.HTTP_400_BAD_REQUEST
```
?

Reply:
Yes, the default comes from the decorator and we only need to set it on the response (and do not need to return the response) if it's not the default.